OvmfPkg/IntelTdx/TdxHelperLib/SecTdxHelper.c

```diff
@@ -523,6 +523,467 @@ IsInValidList (
+      DEBUG ((DEBUG_ERROR, "TdxMetadata : The MemoryAddress and MemoryDataSize must be 4k aligned\n"));
+      return FALSE;
+    }
+
+    switch (Section->Type) {
+      case TDX_METADATA_SECTION_TYPE_BFV:
+        if (Section->Attribute != TDX_METADATA_ATTRIBUTES_EXTENDMR) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The Attribute of BFV Section must be 1\n"));
+          return FALSE;
+        }
+
+        if (Section->RawDataSize == 0) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The RawDataSize of BFV Section must be non-zero\n"));
+          return FALSE;
+        }
+
+        if ((Section->MemoryAddress + Section->MemoryDataSize) != ResetVector ) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The ResetVector should be inside of BFV\n"));
+          return FALSE;
+        }
+
+        NumberOfBFVSection++;
+        break;
+
+      case TDX_METADATA_SECTION_TYPE_CFV:
+        if (Section->Attribute != TDX_METADATA_ATTRIBUTES_EXTENDMR_TD) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The Attribute of CFV Section must be 0\n"));
+          return FALSE;
+        }
+
+        if (Section->RawDataSize == 0) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The RawDataSize of CFV Section must be non-zero\n"));
+          return FALSE;
+        }
+
+        break;
+
+      case TDX_METADATA_SECTION_TYPE_TD_HOB:
+        if (Section->Attribute != TDX_METADATA_ATTRIBUTES_EXTENDMR_TD) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The Attribute of TD HOB Section must be 0\n"));
+          return FALSE;
+        }
+
+        if (Section->RawDataSize != 0) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The RawDataSize of TD HOB Section must be zero\n"));
+          return FALSE;
+        }
+
+        NumberOfTdHobSection++;
+        break;
+
+      case TDX_METADATA_SECTION_TYPE_TEMP_MEM:
+        if (Section->Attribute != TDX_METADATA_ATTRIBUTES_EXTENDMR_TD) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The Attribute of Temp Memory Section must be 0\n"));
+          return FALSE;
+        }
+
+        if (Section->RawDataSize != 0) {
+          DEBUG ((DEBUG_ERROR, "TdxMetadata : The RawDataSize of Temp Memory Section must be zero\n"));
+          return FALSE;
+        }
+
+        break;
+
+      default:
+        DEBUG ((DEBUG_ERROR, "TdxMetadata : Section type is unknown. Type: 0x%08x\n", Section->Type));
+        return FALSE;
+    }
+
+    Address += sizeof (TDX_METADATA_SECTION);
+    Section  = NULL;
+  }
+
+  if (NumberOfBFVSection == 0) {
+    DEBUG ((DEBUG_ERROR, "TdxMetadata : TDVF Metadata shall inclued at least one BFV Section\n"));
+    return FALSE;
+  }
+
+  if (NumberOfTdHobSection != 1 ) {
+    DEBUG ((DEBUG_ERROR, "TdxMetadata : TDVF Metadata shall have only one TD HOB Section\n"));
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+/**
+  This function would check whether the unaccepted memory in the range of
+  Tdx Metadata Temporay Memory.

```

Comment:
Before calling this funtion, the validness of TdxMetadata shall be checked by calling ValidateTdxMetadata.